[docs] assertions refactoring (#4901)
DevExpress/testcafe: docs/articles/documentation/reference/test-api/testcontroller/expect/README.md

---
layout: docs
title: t.expect Method

> permalink: /documentation/reference/test-api/testcontroller/expect/README.html

Reviewer: The link should be:

```
permalink: /documentation/reference/test-api/testcontroller/expect/
```

---
# t.expect() Method

This method accepts the actual value. You can pass a value, a [Selector's DOM node state property](../../../../guides/basic-guides/select-page-elements.md#define-assertion-actual-value)
or a [client function](../../../../guides/basic-guides/obtain-data-from-the-client.md) promise.
TestCafe automatically waits for node state properties to obtain a value and for client functions to execute.

> See [Smart Assertion Query Mechanism](../../../../guides/basic-guides/assert.md#smart-assertion-query-mechanism) for details.

Reviewer: I think we should say that this method should be followed by assertion methods (and even link them all, like in the guide).

We should also say about the value type (as we do in the template).

And add an example.

> You cannot pass a regular promise to the `expect` method unless the options.allowUnawaitedPromise option is enabled.

> > {% include assertions/allowUnawaitedPromise.md %}

Reviewer: But this template contains a header - once we remove it from there, it should be okay.

---
layout: docs
title: t.expect Method
permalink: /documentation/reference/test-api/testcontroller/expect/
---
# t.expect Method

This method accepts the actual value.

You can pass a value, a [Selector's DOM node state property](../../../../guides/basic-guides/select-page-elements.md#define-assertion-actual-value)
or a [client function](../../../../guides/basic-guides/obtain-data-from-the-client.md) promise.
TestCafe automatically waits for node state properties to obtain a value and for client functions to execute. See [Smart Assertion Query Mechanism](../../../../guides/basic-guides/assert.md#smart-assertion-query-mechanism) for details.

> The value passed as the `actual` parameter needs to be of the data type supported by the assertion method.

Reviewer 2: ```suggestion
The value passed as the `actual` parameter needs to an acceptable data type.
```

Author: @dirk-pieterse thank you for the suggestion! But we do need to highlight that the data type must be supported by assertion methods. Simply saying "acceptable data type" is not a good option here

Reviewer 2: When you write it this way, there can be multiple interpretations of "supported by the assertion method".

Reviewer 2: ```suggestion
The value passed as the `actual` parameter should be a data type the assertion method accepts.
```
